### PR TITLE
workers: only send XMLRpc commands to proxy main address

### PR DESCRIPTION
**Pull request by Kaian**
Avoid looping while handling ProxyUsers and ProxyTrunks tables. Only one record is valid in the table.

This was discovered in #856 by @powerpbx and replaces #857 pull request

### CODE REVIEW

File: library/Ivoz/Provider/Domain/Model/ProxyUser/ProxyUserRepository.php [hunk 1]
```diff
@@ -7,5 +7,9 @@
 
 interface ProxyUserRepository extends ObjectRepository, Selectable
 {
-
+    /**
+     * @inheritdoc
+     * @see ProxyUserRepository::getProxyMainAddress
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/ProxyUser/ProxyUserRepository.php]: No need to self reference here :)

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/ProxyTrunkDoctrineRepository.php [hunk 2]
```diff
@@ -19,4 +20,25 @@ public function __construct(RegistryInterface $registry)
...
+
+        $qb
+            ->select('self')
+            ->addCriteria(
+                CriteriaHelper::fromArray([
+                    ['id', 'eq', 1],
+                ])
+            );
+
+        $result = $qb->getQuery()->getResult();
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/ProxyTrunkDoctrineRepository.php]: ->getOneOrNullResult() could be interesting here

**Comment by powerpbx** [comment]
A little confused why you are actively trying to block clustered features.  Perhaps we need to understand the direction of this project better.  We were under the impression it is supposed to be a distributed solution.

We have multiple geographically distributed proxies working now.  We use it as primary, secondary and not for load balancing.  Almost all hard SIP phones are designed for this purpose with primary and secondary server settings.

**Comment by powerpbx** [comment]
Perhaps you can create another table(s) for this purpose.  The code works well as it is now so it would be a shame to lose this functionality.  It seems like a step backwards from the stated goal.

> From its beginning, IvozProvider was designed to be installed distributed between multiple machines...

**Comment by cruzccl** [comment]
Hi @powerpbx,

My 2 cents about this thread:

- The only way to ensure that our IvozProvider development does not diverge from your fork and its modifications is to contribute them via PR so that they are included officially in the main project (after the PR gets the approval of IvozProvider development team). If not, we may introduce changes that require that you readapt your changes or, even worse, that make your changes un-adaptable.

- We are not tied to the changes made by any user using a modified version of IvozProvider: you can fork the project and do whatever you want (as long as you fulfill the license requirements described [here)](https://github.com/irontec/ivozprovider/blob/bleeding/LICENSE), but you cannot control where the original project goes.

- Note that the very same thing happens to us with our custom patches on the different softwares shipped with IvozProvider (Asterisk, Kamailio, rtpengine, CGRateS...): if our changes are not accepted or are not suitable for the original project, we have to readapt our patches every time we need changes introduced in the main project (e.g. CGRateS sentinel support in #828).

- This way to work is fair for the original project too, that gets enhancements from contributors that prefer contributing to the project instead of mantaining their patches (leaving aside the ethical fact of contributing to the people that shared their entire work on the first time).

All the above is the way I think about free software projects collaboration and I hope you agree with me.

Leaving all this aside, I have several technical doubts about your multi-proxy implementation and I will be pleased to talk about them in a new issue or in #649.

Regards and thank you for your thoughts and time,

**Comment by powerpbx** [comment]
You should update your documentation if you don't plan on supporting multiple proxies.  Those images showing multiple proxy blocks is not representative of what this product is.

**Pull request merged**